test/position-managers/IncreaseLiquidity.t.sol

```diff
@@ -0,0 +1,327 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.24;
+
+import "forge-std/Test.sol";
+import {GasSnapshot} from "forge-gas-snapshot/GasSnapshot.sol";
+import {PoolManager} from "@uniswap/v4-core/src/PoolManager.sol";
+import {IPoolManager} from "@uniswap/v4-core/src/interfaces/IPoolManager.sol";
+import {IHooks} from "@uniswap/v4-core/src/interfaces/IHooks.sol";
+import {Deployers} from "@uniswap/v4-core/test/utils/Deployers.sol";
+import {Currency, CurrencyLibrary} from "@uniswap/v4-core/src/types/Currency.sol";
+import {PoolId, PoolIdLibrary} from "@uniswap/v4-core/src/types/PoolId.sol";
+import {PoolKey} from "@uniswap/v4-core/src/types/PoolKey.sol";
+import {BalanceDelta, toBalanceDelta} from "@uniswap/v4-core/src/types/BalanceDelta.sol";
+import {PoolSwapTest} from "@uniswap/v4-core/src/test/PoolSwapTest.sol";
+import {LiquidityAmounts} from "@uniswap/v4-core/test/utils/LiquidityAmounts.sol";
+import {TickMath} from "@uniswap/v4-core/src/libraries/TickMath.sol";
+import {FixedPointMathLib} from "solmate/utils/FixedPointMathLib.sol";
+import {StateLibrary} from "@uniswap/v4-core/src/libraries/StateLibrary.sol";
+import {Fuzzers} from "@uniswap/v4-core/src/test/Fuzzers.sol";
+
+import {IERC20} from "forge-std/interfaces/IERC20.sol";
+import {ERC721} from "@openzeppelin/contracts/token/ERC721/ERC721.sol";
+
+import {PositionManager} from "../../src/PositionManager.sol";
+import {LiquidityRange, LiquidityRangeId, LiquidityRangeIdLibrary} from "../../src/types/LiquidityRange.sol";
+import {Actions, IPositionManager} from "../../src/interfaces/IPositionManager.sol";
+import {LiquidityOperations} from "../shared/LiquidityOperations.sol";
+import {Planner} from "../utils/Planner.sol";
+import {FeeMath} from "../shared/FeeMath.sol";
+
+import "forge-std/console2.sol";
+
+contract IncreaseLiquidityTest is Test, Deployers, GasSnapshot, Fuzzers, LiquidityOperations {
+    using FixedPointMathLib for uint256;
+    using CurrencyLibrary for Currency;
+    using LiquidityRangeIdLibrary for LiquidityRange;
+    using PoolIdLibrary for PoolKey;
+    using Planner for Planner.Plan;
+    using FeeMath for IPositionManager;
+
+    PoolId poolId;
+    address alice = makeAddr("ALICE");
+    address bob = makeAddr("BOB");
+
+    uint256 constant STARTING_USER_BALANCE = 10_000_000 ether;
+
+    // expresses the fee as a wad (i.e. 3000 = 0.003e18 = 0.30%)
+    uint256 FEE_WAD;
+
+    LiquidityRange range;
+
+    // Error tolerance.
+    uint256 tolerance = 0.00000000001 ether;
```

Comment:
Please can you add to the PR TODO list that tolerance/off by 1 errors in tests need to be fixed

Reply:
all the tests that use tolerance are expected & i went through and fixed some of the weirder off by 1s. there is only 1 off by one test that I will play around with. I think if we use getFeesOwed it might fix it